app/utils/createEntityReducer.js

```diff
@@ -99,11 +106,38 @@ export function entities(key: string, fetchType?: ?AsyncActionType) {
   };
 }
 
-export function optimistic(mutateType?: ?AsyncActionType) {
+/* This part of CER will handle deletion of values of a given entity.
+ * For actions of the type given into `types.delete` when setting up CER will be deleted.
+ *
+ * Make sure to set `meta.id` in the action.
+ */
+export function deleteEntities(deleteTypes: ?EntityReducerTypes) {
+  return (state: any = defaultState, action: any) => {
+    if (
+      !toArray(deleteTypes).some(
+        deleteType => action.type === deleteType.SUCCESS
+      )
+    ) {
+      return state;
+    }
+    const resultId = action.meta && action.meta.id;
+
+    if (!resultId) return state;
+
+    return {
+      ...state,
+      byId: omit(state.byId, resultId),
```

Comment:
Should we do this, or just keep the element in the map? Most reducers only removed the entity from the "items" list. We have no track on the relations, so it is impossible to know if the pk is refrenced in another object... :thinking:

Reply:
Good question. Not sure if there is any reason for the current behaviour. I think deleting it everywhere makes sense. If there is UI using the deleted object it should probably crash and let us know instead of keep working.

Reply:
:+1: Ill just keep it like this this. Agree that this makes more sense than the current behavior